diff --git a/dist/esm/core/router-generator-plugin.js b/dist/esm/core/router-generator-plugin.js
index ce5223bd9f71e7fa2fcdc6b95f4d1e95ed5e03ea..ed7ed4ca95f4f7119c91473c8cf74aad1875d91c 100644
--- a/dist/esm/core/router-generator-plugin.js
+++ b/dist/esm/core/router-generator-plugin.js
@@ -46,23 +46,24 @@ const unpluginRouterGeneratorFactory = (options = {}) => {
         event
       });
     },
-    async buildStart() {
-      await generate();
-    },
+    // async buildStart() {
+    //   await generate();
+    // },
     vite: {
-      configResolved() {
+      async configResolved() {
         initConfigAndGenerator();
+        await generate()
       },
-      applyToEnvironment(environment) {
-        if (userConfig.plugin?.vite?.environmentName) {
-          return userConfig.plugin.vite.environmentName === environment.name;
-        }
-        return true;
-      },
-      async buildStart() {
-        await generate();
-      },
-      sharedDuringBuild: true
+      // applyToEnvironment(environment) {
+      //   if (userConfig.plugin?.vite?.environmentName) {
+      //     return userConfig.plugin.vite.environmentName === environment.name;
+      //   }
+      //   return true;
+      // },
+      // async buildStart() {
+      //   await generate();
+      // },
+      // sharedDuringBuild: true
     },
     rspack(compiler) {
       initConfigAndGenerator();
